File: backend/shop-floor-assistance-backend/src/main/java/com/shopfloor/backend/api/transferobjects/TaskTO.java
package com.shopfloor.backend.api.transferobjects;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Getter
@Setter
public class TaskTO {

    private Long id;

    private String name;

    private String description;

    private Long createdBy;

    private Long updatedBy;

    private Date createdAt;

    private Date updatedAt;

    private List<ItemTO> items;

    public TaskTO() {
        this.items = new ArrayList<ItemTO>();
    }
}

File: frontend/shop-floor-assistance-frontend/src/app/types/taskTO.ts
import { workflowTO } from "./workflowTO"; 
import { itemTO } from "./itemTO";

export type taskTO = {
  id?: number;
  name: string;
  description?: string;

  createdBy?: number;
  updatedBy?: number;
  createdAt?: Date;
  updatedAt?: Date;

  items: itemTO[]; 
};
